src/tests/roc_pipeline/test_helpers/utils.h

```diff
@@ -24,7 +25,8 @@ namespace {
 
 const double SampleEpsilon = 0.00001;
 
-const core::nanoseconds_t TimestampEpsilon = core::Microsecond;
+const core::nanoseconds_t TimestampEpsilon =
+    (core::nanoseconds_t)(1. / DefaultSampleRate * core::Second);
```

Comment:
Different tests may use different sample rates.

We can rework expect_capture_timestamp() so that it accepts sample_spec and uses it to calculate epsilon. Or hard-code rate-independent epsilon that is just large enough for all rates.